### PR TITLE
[CMS-596] Add new drupal-9-legacy-project-to-pantheon docs page.

### PR DESCRIPTION
**Pull request by kporras07**
Closes CMS-596

## Summary

**[Migrating a non composer managed Drupal 9 site to Pantheon](https://pr-7146--pantheon-docs.my.pantheonfrontend.website/docs/guides/drupal-9-legacy-project-to-pantheon)** - Create documentation page.

**Release**:
- [X] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-7146](https://pr-7146--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `56.99999999999999` | `61` | `4` |
| Best Practice Score | `79` | `79` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-7146` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/34348/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/34348/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-7146--pantheon-docs.my.pantheonfrontend.website/docs)

**Comment by kporras07** [comment]
Please note in the preview link that the screenshots for the DB import process are actually Wordpress screenshots. Not sure if that matters...

File: source/content/guides/drupal-9-legacy-project-to-pantheon.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  # From $DESTINATION:
+  git add config
+  git commit -m "Pull in configuration from source site"
+  ```
+
+If no files are copied through this step, that's acceptable.
+
+## Add Custom and Contrib Code
+
+In your new project structure, replicate your selection of contributed modules and themes, and any custom modules or themes your development team has created.
```
Comment (greg-1-anderson) [inline source/content/guides/drupal-9-legacy-project-to-pantheon.md:85]: Perhaps clarify that this section provides instructions on how to do this.

File: source/content/guides/drupal-9-legacy-project-to-pantheon.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+The goal of this process is to have Composer manage all the site's contrib modules, contrib themes, core upgrades, and libraries (referred to as _contributed code_). The only things that should be migrated from the existing site are custom code, custom themes, and custom modules that are specific to the existing site.
+
+
+#### Modules and Themes
+
+
+The steps here ensure that any modules and themes from [drupal.org](https://drupal.org) are in the `composer.json` `require` list.
+
+Once Composer is aware of all the contributed code, you'll be able to run `composer update` from within the directory to have Composer upgrade all the contributed code automatically.
```
Comment (greg-1-anderson) [inline source/content/guides/drupal-9-legacy-project-to-pantheon.md:97]: The Pantheon dashboard will also update Composer dependencies in addition to updating the files from the upstream.

File: source/content/guides/drupal-9-legacy-project-to-pantheon.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+Begin by reviewing the existing site's code. Check for contributed modules in `/modules`, `/modules/contrib`, `/sites/all/modules`, and `/sites/all/modules/contrib`.
+
+1. Review the site and make a list of exactly what versions of modules and themes you depend on. One way to do this is to run the `pm:list` Drush command from within a contributed modules folder (e.g. `/modules`, `/themes`, `/themes/contrib`, `/sites/all/themes`, `/sites/all/themes/contrib`, etc.).
+
+  This will list each module followed by the version of that module that is installed:
+
+  ```bash{promptUser:user}
+  drush pm:list --no-core --fields=name,version  --format=table
```
Comment (greg-1-anderson) [inline source/content/guides/drupal-9-legacy-project-to-pantheon.md:106]: This command should instead be:
```
drush pm:list --no-core --fields=name,project,version  --format=table
```

File: source/content/guides/drupal-9-legacy-project-to-pantheon.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+  ```
+
+1. Add these modules to your new codebase using Composer by running the following for each module in the `$DESTINATION` directory:
+
+  ```bash{promptUser:user}
+  composer require drupal/MODULE_NAME:^VERSION
+  ```
+
+  Where `MODULE_NAME` is the machine name of the module in question, and `VERSION` is the version of that module the site is currently using. Composer may pull in a newer version than what you specify, depending upon what versions are available. You can read more about the caret (`^`) in the [Composer documentation](https://getcomposer.org/doc/articles/versions.md#caret-version-range-).
```
Comment (greg-1-anderson) [inline source/content/guides/drupal-9-legacy-project-to-pantheon.md:115]: This should not be the module name; it should be the project name from the "project" field.

Comment (greg-1-anderson) [reply source/content/guides/drupal-9-legacy-project-to-pantheon.md:115]: This is also wrong on https://pantheon.io/docs/guides/composer-convert; I thought that we had corrected it in the past, but the project / module confusion has returned in the latest version.

**Comment by wordsmither** [comment]
lgmt

**Pull request merged**